cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -389,26 +389,33 @@ externalSetupMethod verbosity options pkg bt mkargs = do
       Nothing    -> getInstalledPackages verbosity
                     comp (usePackageDB options') conf
 
-  cabalLibVersionToUse :: IO (Version, (Maybe ComponentId)
+  cabalLibVersionToUse :: IO (Version, Maybe ComponentId
                              ,SetupScriptOptions)
   cabalLibVersionToUse =
-    case useCabalSpecVersion options of
-      Just version -> do
+    case find (hasCabal . snd) (useDependencies options) of
```

Comment:
@23Skidoo The [comment](https://github.com/haskell/cabal/blob/c4fc3bde3b2690063a63f65de9787f82bc9bef26/cabal-install/Distribution/Client/SetupWrapper.hs#L134) over the `useCabalVersion` field says that it should be used when `useDependenciesExclusive` is not set, so it isn't always set by the dependency solver.  I used `useDependencies` because SetupWrapper already uses it to check whether the solver has picked a Cabal version: https://github.com/haskell/cabal/blob/c4fc3bde3b2690063a63f65de9787f82bc9bef26/cabal-install/Distribution/Client/SetupWrapper.hs#L629.